drivers/kscan/kscan_gt911.c

```diff
@@ -96,22 +96,29 @@ static int gt911_process(const struct device *dev)
 		return 0;
 	}
 
-	/* need to clear the status */
-	uint8_t clear_buffer[3] = {(uint8_t)REG_STATUS, (uint8_t)(REG_STATUS >> 8), 0};
-
-	r = i2c_write_dt(&config->bus, clear_buffer, sizeof(clear_buffer));
-	if (r < 0) {
-		return r;
-	}
-
-	/* obtain first point X, Y coordinates and event from:
-	 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
-	 */
-	reg_addr = REG_FIRST_POINT;
-	r = i2c_write_read_dt(&config->bus, &reg_addr, sizeof(reg_addr),
-							&pointRegs, sizeof(pointRegs));
-	if (r < 0) {
-		return r;
+	if (status & TOUCH_STATUS_MSK) {
+		/* need to clear the status */
+		uint8_t clear_buffer[3] = {(uint8_t)REG_STATUS,
```

Comment:
@danieldegrasse not familiar with this chip is ok to clear rest of the bits in this register or should this be read and write operation?

Reply:
Per programing guide:  https://www.crystalfontz.com/controllers/GOODIX/GT911ProgrammingGuide/478, The status register at 0x814E states the following about the TOUCH_STATUS_MSK bit: "After reading coordinates, host should configure this flag (or the entire byte) to 0 via I2 C." . So clearing this byte is valid, since we read the coordinates in the same function.